pennylane/tape/tape.py

```diff
@@ -193,13 +193,18 @@ def expand_tape(tape, depth=1, stop_at=None, expand_measurements=False):
 
             if isinstance(obj, (qml.operation.Operation, qml.measure.MeasurementProcess)):
                 # Object is an operation; query it for its expansion
-                try:
-                    obj = obj.expand()
-                except NotImplementedError:
-                    # Object does not define an expansion; treat this as
-                    # a stopping condition.
+                # try:
+                expanded_obj = obj.expand()
+                if expanded_obj is None:
                     getattr(new_tape, queue).append(obj)
                     continue
+                else:
+                    obj = expanded_obj
```

Comment:
I must admit, without being able to use the new walrus operator from Python 3.8, the old EAFP approach looks slightly cleaner to my eye

Reply:
In terms of readability, I prefer the new "look before you leap" approach. 
This approach will also require fewer changes once we have a system to check whether or not `compute` methods are implemented. Then we could do something here like:
```
if not hasattr(obj, "decomposition"):
```